### ISSUE TITLE
[mono] instructions for running tests on windows using Mono with release/6.0 branch

### ISSUE THREAD

**Issue by srxqds** [dotnet/runtime#63189]
https://github.com/dotnet/runtime/blob/release/6.0/docs/workflow/testing/mono/testing.md 
I do as the testing.md steps. 

> ./build.cmd -excludemonofailures release/debug
> ../../dotnet.cmd msbuild /t:RunCoreClrTests /p:CoreClrTestConfig=Debug /p:LibrariesTestConfig=Release /p:Configuration=Release mono.proj

the first command can built tests success, but the second command to run test always run failed:

![image](https://user-images.githubusercontent.com/5187024/147660191-67a51660-e827-48d4-8510-7650e5658eeb.png)

what steps may be left to build, is there any detail document?

**Comment by srxqds** [comment]
I try again this day, fail always,

this is test command and the first output:

 ../../dotnet.cmd msbuild /t:RunCoreClrTests /p:CoreClrTestConfig=Debug /p:LibrariesTestConfig=Debug /p:Configuration=Debug mono.proj
C:\Program Files\dotnet
用于 .NET 的 Microsoft (R) 生成引擎版本 17.0.0+c9eb9dd64
版权所有(C) Microsoft Corporation。保留所有权利。

  Copying Mono Runtime artifacts from 'E:\work\netease-gitlab\dotnet\runtime\artifacts\bin\mono\windows.x64.Debug\' to 'E:\work\netease-gitlab\dotnet\runtime\artifacts\tests\coreclr\windows.x64.Debug\Tests\Core_Root\'.'
  "C:\Users\qiudesuo\AppData\Local\Programs\Python\Python310\python.exe" "E:\work\netease-gitlab\dotnet\runtime\src\tests\..\..\src\tests\run.py" -arch x64 -build_type Debug
  host_os                  : windows
  arch                     : x64
  build_type               : Debug
  runtime_repo_location    : E:\work\netease-gitlab\dotnet\runtime
  core_root                : E:\work\netease-gitlab\dotnet\runtime\artifacts\tests\coreclr\windows.x64.Debug\Tests\Core_Root
  test_location            : E:\work\netease-gitlab\dotnet\runtime\artifacts\tests\coreclr\windows.x64.Debug
  Setting __TestTimeout=900000
  Setting CORE_ROOT=E:\work\netease-gitlab\dotnet\runtime\artifacts\tests\coreclr\windows.x64.Debug\Tests\Core_Root
  E:\work\netease-gitlab\dotnet\runtime\dotnet.cmd msbuild E:\work\netease-gitlab\dotnet\runtime\src\tests\run.proj /p:Runtests=true /clp:showcommandline /fileloggerparameters:"Verbosity=normal;LogFile=E:\work\netease-gitlab\dotnet\runtime\artifacts\log\TestRunResults_windows_x64_Debug.log" /fileloggerparameters1:"WarningsOnly;LogFile=E:\work\netease-gitlab\dotnet\runtime\artifacts\log\TestRunResults_windows_x64_Debug.wrn" /fileloggerparameters2:"ErrorsOnly;LogFile=E:\work\netease-gitlab\dotnet\runtime\artifacts\log\TestRunResults_windows_x64_Debug.err" /consoleloggerparameters:Summary /p:TargetOS=windows /p:TargetArchitecture=x64 /p:Configuration=Debug /p:__LogsDir=E:\work\netease-gitlab\dotnet\runtime\artifacts\log /bl:E:\work\netease-gitlab\dotnet\runtime\artifacts\log\TestRunResults_windows_x64_Debug.binlog
  C:\Program Files\dotnet
  用于 .NET 的 Microsoft (R) 生成引擎版本 17.0.0+c9eb9dd64
  版权所有(C) Microsoft Corporation。保留所有权利。

  C:\Program Files\dotnet\sdk\6.0.101\MSBuild.dll -distributedlogger:Microsoft.DotNet.Tools.MSBuild.MSBuildLogger,C:\Program Files\dotnet\sdk\6.0.101\dotnet.dll*Microsoft.DotNet.Tools.MSBuild.MSBuildForwardingLogger,C:\Program Files\dotnet\sdk\6.0.101\dotnet.dll -maxcpucount -verbosity:m /bl:E:\work\netease-gitlab\dotnet\runtime\artifacts\log\TestRunResults_windows_x64_Debug.binlog /clp:showcommandline /consoleloggerparameters:Summary /fileloggerparameters1:WarningsOnly;LogFile=E:\work\netease-gitlab\dotnet\runtime\artifacts\log\TestRunResults_windows_x64_Debug.wrn /fileloggerparameters2:ErrorsOnly;LogFile=E:\work\netease-gitlab\dotnet\runtime\artifacts\log\TestRunResults_windows_x64_Debug.err /fileloggerparameters:Verbosity=normal;LogFile=E:\work\netease-gitlab\dotnet\runtime\artifacts\log\TestRunResults_windows_x64_Debug.log /p:Runtests=true /p:TargetOS=windows /p:TargetArchitecture=x64 /p:Configuration=Debug /p:__LogsDir=E:\work\netease-gitlab\dotnet\runtime\artifacts\log E:\work\netease-gitlab\dotnet\runtime\src\tests\run.proj
    E:\work\netease-gitlab\dotnet\runtime\artifacts\tests\coreclr\windows.x64.Debug\Tests\Core_Root\corerun.exe E:\work\netease-gitlab\dotnet\runtime\artifacts\tests\coreclr\windows.x64.Debug\Tests\Core_Root\xunit\xunit.console.dll .\baseservices\callconvs\baseservices.callconvs.XUnitWrapper.dll .\baseservices\compilerservices\baseservices.compilerservices.XUnitWrapper.dll .\baseservices\exceptions\baseservices.exceptions.XUnitWrapper.dll .\baseservices\mono\baseservices.mono.XUnitWrapper.dll .\baseservices\threading\baseservices.threading.XUnitWrapper.dll .\baseservices\TieredCompilation\baseservices.TieredCompilation.XUnitWrapper.dll .\baseservices\typeequivalence\baseservices.typeequivalence.XUnitWrapper.dll .\baseservices\varargs\baseservices.varargs.XUnitWrapper.dll .\CoreMangLib\system\CoreMangLib.system.XUnitWrapper.dll .\Exceptions\ForeignThread\Exceptions.ForeignThread.XUnitWrapper.dll .\GC\API\GC.API.XUnitWrapper.dll .\GC\Coverage\GC.Coverage.XUnitWrapper.dll .\GC\Features\GC.Features.XUnitWrapper.dll .\GC\LargeMemory\GC.LargeMemory.XUnitWrapper.dll .\GC\Regressions\GC.Regressions.XUnitWrapper.dll .\GC\Scenarios\GC.Scenarios.XUnitWrapper.dll .\GC\Stress\GC.Stress.XUnitWrapper.dll .\ilasm\PortablePdb\ilasm.PortablePdb.XUnitWrapper.dll .\ilasm\System\ilasm.System.XUnitWrapper.dll .\Interop\ArrayMarshalling\Interop.ArrayMarshalling.XUnitWrapper.dll .\Interop\DllImportAttribute\Interop.DllImportAttribute.XUnitWrapper.dll .\Interop\ExecInDefAppDom\Interop.ExecInDefAppDom.XUnitWrapper.dll .\Interop\ICastable\Interop.ICastable.XUnitWrapper.dll .\Interop\ICustomMarshaler\Interop.ICustomMarshaler.XUnitWrapper.dll .\Interop\IJW\Interop.IJW.XUnitWrapper.dll .\Interop\LayoutClass\Interop.LayoutClass.XUnitWrapper.dll .\Interop\MarshalAPI\Interop.MarshalAPI.XUnitWrapper.dll .\Interop\NativeLibrary\Interop.NativeLibrary.XUnitWrapper.dll .\Interop\PInvoke\Interop.PInvoke.XUnitWrapper.dll .\Interop\StringMarshalling\Interop.StringMarshalling.XUnitWrapper.dll .\Interop\StructMarshalling\Interop.StructMarshalling.XUnitWrapper.dll .\Interop\StructPacking\Interop.StructPacking.XUnitWrapper.dll .\Interop\SuppressGCTransition\Interop.SuppressGCTransition.XUnitWrapper.dll .\Interop\UnmanagedCallConv\Interop.UnmanagedCallConv.XUnitWrapper.dll .\Interop\UnmanagedCallersOnly\Interop.UnmanagedCallersOnly.XUnitWrapper.dll .\Interop\WinRT\Interop.WinRT.XUnitWrapper.dll .\JIT\CheckProjects\JIT.CheckProjects.XUnitWrapper.dll .\JIT\CodeGenBringUpTests\JIT.CodeGenBringUpTests.XUnitWrapper.dll .\JIT\Directed\JIT.Directed.XUnitWrapper.dll .\JIT\Generics\JIT.Generics.XUnitWrapper.dll .\JIT\HardwareIntrinsics\JIT.HardwareIntrinsics.XUnitWrapper.dll .\JIT\IL_Conformance\JIT.IL_Conformance.XUnitWrapper.dll .\JIT\Intrinsics\JIT.Intrinsics.XUnitWrapper.dll .\JIT\jit64\JIT.jit64.XUnitWrapper.dll .\JIT\Math\JIT.Math.XUnitWrapper.dll .\JIT\Methodical\JIT.Methodical.XUnitWrapper.dll .\JIT\opt\JIT.opt.XUnitWrapper.dll .\JIT\Performance\JIT.Performance.XUnitWrapper.dll .\JIT\Regression\JIT.Regression.XUnitWrapper.dll .\JIT\RyuJIT\JIT.RyuJIT.XUnitWrapper.dll .\JIT\SIMD\JIT.SIMD.XUnitWrapper.dll .\JIT\Stress\JIT.Stress.XUnitWrapper.dll .\JIT\superpmi\JIT.superpmi.XUnitWrapper.dll .\Loader\AssemblyDependencyResolver\Loader.AssemblyDependencyResolver.XUnitWrapper.dll .\Loader\AssemblyLoadContext30Extensions\Loader.AssemblyLoadContext30Extensions.XUnitWrapper.dll .\Loader\binding\Loader.binding.XUnitWrapper.dll .\Loader\classloader\Loader.classloader.XUnitWrapper.dll .\Loader\CollectibleAssemblies\Loader.CollectibleAssemblies.XUnitWrapper.dll .\Loader\ContextualReflection\Loader.ContextualReflection.XUnitWrapper.dll .\Loader\regressions\Loader.regressions.XUnitWrapper.dll .\managed\Compilation\managed.Compilation.XUnitWrapper.dll .\profiler\elt\profiler.elt.XUnitWrapper.dll .\profiler\eventpipe\profiler.eventpipe.XUnitWrapper.dll .\profiler\gc\profiler.gc.XUnitWrapper.dll .\profiler\multiple\profiler.multiple.XUnitWrapper.dll .\profiler\rejit\profiler.rejit.XUnitWrapper.dll .\profiler\transitions\profiler.transitions.XUnitWrapper.dll .\profiler\unittest\profiler.unittest.XUnitWrapper.dll .\readytorun\crossgen2\readytorun.crossgen2.XUnitWrapper.dll .\readytorun\DynamicMethodGCStress\readytorun.DynamicMethodGCStress.XUnitWrapper.dll .\readytorun\multifolder\readytorun.multifolder.XUnitWrapper.dll .\readytorun\r2rdump\readytorun.r2rdump.XUnitWrapper.dll .\readytorun\tests\readytorun.tests.XUnitWrapper.dll .\reflection\DefaultInterfaceMethods\reflection.DefaultInterfaceMethods.XUnitWrapper.dll .\reflection\GenericAttribute\reflection.GenericAttribute.XUnitWrapper.dll .\reflection\Modifiers\reflection.Modifiers.XUnitWrapper.dll .\reflection\RefEmit\reflection.RefEmit.XUnitWrapper.dll .\reflection\regression\reflection.regression.XUnitWrapper.dll .\reflection\SetValue\reflection.SetValue.XUnitWrapper.dll .\reflection\StaticInterfaceMembers\reflection.StaticInterfaceMembers.XUnitWrapper.dll .\Regressions\coreclr\Regressions.coreclr.XUnitWrapper.dll .\tracing\eventactivityidcontrol\tracing.eventactivityidcontrol.XUnitWrapper.dll .\tracing\eventcounter\tracing.eventcounter.XUnitWrapper.dll .\tracing\eventlistener\tracing.eventlistener.XUnitWrapper.dll .\tracing\eventpipe\tracing.eventpipe.XUnitWrapper.dll -parallel collections -html E:\work\netease-gitlab\dotnet\runtime\artifacts\log\TestRun.html -xml E:\work\netease-gitlab\dotnet\runtime\artifacts\log\TestRun.xml  -notrait category=outerloop -notrait category=failing -nocolor

**Comment by srxqds** [comment]
> /cc @lateralusX @steveisok Is there anything special that needs to be done to run the runtime tests on mono on Windows?

hi, @lateralusX @steveisok can you help me?

**Comment by steveisok** [comment]
@lateralusX I took a look and I'm not sure what the problem is.

**Comment by srxqds** [comment]
hi, how can pr action run test, but i can't found the command in the pr action script?

anyone can help me? I want run all tests and fixed some bug.

**Comment by steveisok** [comment]
Looking at this further, it seems the test host is crashing.  An access violation at https://github.com/dotnet/runtime/blob/bd6a64bc5275e47a69aa6a42eab10e1f3bf3faf5/src/mono/mono/metadata/property-bag.c#L25

Stack trace:

```
coreclr.dll!mono_property_bag_get(MonoPropertyBag * bag, int tag) Line 25
	at C:\dev\runtime\src\mono\mono\metadata\property-bag.c(25)
coreclr.dll!mono_class_get_marshal_info(_MonoClass * klass) Line 268
	at C:\dev\runtime\src\mono\mono\metadata\class-accessors.c(268)
coreclr.dll!mono_marshal_load_type_info(_MonoClass * klass) Line 5278
	at C:\dev\runtime\src\mono\mono\metadata\marshal.c(5278)
coreclr.dll!mono_struct_delete_old(_MonoClass * klass, char * ptr) Line 5142
	at C:\dev\runtime\src\mono\mono\metadata\marshal.c(5142)
coreclr.dll!mono_struct_delete_old(_MonoClass * klass, char * ptr) Line 5161
	at C:\dev\runtime\src\mono\mono\metadata\marshal.c(5161)
[External Code]
coreclr.dll!mono_jit_runtime_invoke(_MonoMethod * method, void * obj, void * * params, _MonoObject * * exc, _MonoErrorInternal * error) Line 3513
	at C:\dev\runtime\src\mono\mono\mini\mini-runtime.c(3513)
coreclr.dll!do_runtime_invoke(_MonoMethod * method, void * obj, void * * params, _MonoObject * * exc, _MonoErrorInternal * error) Line 2395
	at C:\dev\runtime\src\mono\mono\metadata\object.c(2395)
coreclr.dll!mono_runtime_invoke_checked(_MonoMethod * method, void * obj, void * * params, _MonoErrorInternal * error) Line 2564
	at C:\dev\runtime\src\mono\mono\metadata\object.c(2564)
coreclr.dll!mono_runtime_try_invoke_span(_MonoMethod * method, void * obj, MonoSpanOfObjects * params_span, _MonoObject * * exc, _MonoErrorInternal * error) Line 4772
	at C:\dev\runtime\src\mono\mono\metadata\object.c(4772)
coreclr.dll!mono_runtime_invoke_span_checked(_MonoMethod * method, void * obj, MonoSpanOfObjects * params, _MonoErrorInternal * error) Line 4864
	at C:\dev\runtime\src\mono\mono\metadata\object.c(4864)
coreclr.dll!ves_icall_InternalInvoke(_MonoReflectionMethodHandle method_handle, _MonoObjectHandle this_arg_handle, MonoSpanOfObjects * params_span, _MonoExceptionHandle exception_out, _MonoErrorInternal * error) Line 3480
	at C:\dev\runtime\src\mono\mono\metadata\icall.c(3480)
coreclr.dll!ves_icall_InternalInvoke_raw(volatile void * a0, volatile void * a1, MonoSpanOfObjects * a2, volatile void * a3) Line 255
	at C:\dev\runtime\src\mono\mono\metadata\icall-def-netcore.h(255)
[External Code]
coreclr.dll!mono_jit_runtime_invoke(_MonoMethod * method, void * obj, void * * params, _MonoObject * * exc, _MonoErrorInternal * error) Line 3513
	at C:\dev\runtime\src\mono\mono\mini\mini-runtime.c(3513)
coreclr.dll!do_runtime_invoke(_MonoMethod * method, void * obj, void * * params, _MonoObject * * exc, _MonoErrorInternal * error) Line 2395
	at C:\dev\runtime\src\mono\mono\metadata\object.c(2395)
coreclr.dll!mono_runtime_invoke_checked(_MonoMethod * method, void * obj, void * * params, _MonoErrorInternal * error) Line 2564
	at C:\dev\runtime\src\mono\mono\metadata\object.c(2564)
coreclr.dll!start_wrapper_internal(StartInfo * start_info, unsigned __int64 * stack_ptr) Line 1213
	at C:\dev\runtime\src\mono\mono\metadata\threads.c(1213)
coreclr.dll!start_wrapper(void * data) Line 1262
	at C:\dev\runtime\src\mono\mono\metadata\threads.c(1262)
[External Code]
```

**Comment by lambdageek** [comment]
The System.Drawing.Common test is failing around here: 

https://github.com/dotnet/runtime/blob/bd6a64bc5275e47a69aa6a42eab10e1f3bf3faf5/src/mono/mono/metadata/marshal.c#L5160

The field in question is `System.Drawing.Imaging.MetafileHeaderEmf:EmfHeader` which has type `Nullable<SafeNativeMethods.ENHMETAHEADER>`.  So `ftype->data.klass` is wrong (it's a generic instance, so it should be `mono_class_from_mono_type_internal (ftype)` because `ftype->data.generic_class` is the active field in the union).  Although in this case `ftype->data.generic_class` would be NULL too, it's a union!  So there might be something else going on.

This is _probably_ not really Windows specific - seems like a genuine bug in the marshalling code.

---

If the runtime tests fail the same way, we should repurpose this issue to fix the marshalling. Otherwise we should open a new issue.

**Comment by steveisok** [comment]
The runtime tests seem to fail differently.  The first one I tried was `ThisCallTest`.  The debugger broke on this line https://github.com/dotnet/runtime/blob/ec5471c248a944ff7a1d642c1200d49ab349bef1/src/tests/JIT/Directed/callconv/InstanceCallConvTest.cpp#L46

The stack trace is:

```
>	ThisCallNative.dll!C::GetSize(int __formal) Line 46	C++
 	[External Code]	
 	coreclr.dll!mono_jit_runtime_invoke(_MonoMethod * method, void * obj, void * * params, _MonoObject * * exc, _MonoErrorInternal * error) Line 3513	C
 	coreclr.dll!do_runtime_invoke(_MonoMethod * method, void * obj, void * * params, _MonoObject * * exc, _MonoErrorInternal * error) Line 2395	C
 	coreclr.dll!mono_runtime_invoke_checked(_MonoMethod * method, void * obj, void * * params, _MonoErrorInternal * error) Line 2564	C
 	coreclr.dll!do_exec_main_checked(_MonoMethod * method, _MonoArray * args, _MonoErrorInternal * error) Line 4402	C
 	coreclr.dll!mono_runtime_exec_main_checked(_MonoMethod * method, _MonoArray * args, _MonoErrorInternal * error) Line 4507	C
 	coreclr.dll!mono_runtime_run_main_checked(_MonoMethod * method, int argc, char * * argv, _MonoErrorInternal * error) Line 4071	C
 	coreclr.dll!mono_jit_exec_internal(_MonoDomain * domain, _MonoAssembly * assembly, int argc, char * * argv) Line 1379	C
 	coreclr.dll!mono_jit_exec(_MonoDomain * domain, _MonoAssembly * assembly, int argc, char * * argv) Line 1324	C
 	coreclr.dll!main_thread_handler(void * user_data) Line 1458	C
 	coreclr.dll!mono_main(int argc, char * * argv) Line 2677	C
 	coreclr.dll!monovm_execute_assembly(int argc, const char * * argv, const char * managedAssemblyPath, unsigned int * exitCode) Line 268	C
 	coreclr.dll!coreclr_execute_assembly(void * hostHandle, unsigned int domainId, int argc, const char * * argv, const char * managedAssemblyPath, unsigned int * exitCode) Line 82	C
 	corerun.exe!run(const configuration & config) Line 368	C++
 	corerun.exe!wmain(const int argc, const wchar_t * * argv) Line 563	C++
```

I'll look for others to see if they are the same.

**Comment by lambdageek** [comment]
ok, so to answer @srxqds: the infrastructure for running libraries and runtime tests is working.  Some of the tests, however, are currently failing on Windows.   This is expected.  We just started building .NET 7 Mono on Windows a few weeks ago - we haven't started to work on the bugs.  We will open issues for the tests that fail.

**Comment by srxqds** [comment]
ok, thank you, so the tests will pass on macos/ios android?

**Comment by lateralusX** [comment]
The CoreCLR runtime tests have never been running on Mono Windows on CI (but libraries tests do). I had a PR for it a while back, trying to get it at least into the staging lane, https://github.com/dotnet/runtime/pull/47526, but since it needed additional work (mainly walking through and evaluate failures and exclude tests), it was put on hold until we had some time to do that work, so there will be failures when running the runtime tests as is, I expect most failures to be similar to other Mono platforms, except they either don't run on other platforms or have already been excluded.

CoreCLR runtime tests is currently running on CI for Mono Linux/MacOS as well as iOS/Android.

**Comment by srxqds** [comment]
thank you for reply

**Comment by lateralusX** [comment]
Run through the test suite on Mono Windows and we get the following result:

```
       baseservices.callconvs.XUnitWrapper.dll                  Total:    0
       baseservices.compilerservices.XUnitWrapper               Total:    2, Errors: 0, Failed:  0, Skipped: 0, Time:   0,232s
       baseservices.exceptions.XUnitWrapper.dll                 Total:    0
       baseservices.mono.XUnitWrapper                           Total:    1, Errors: 0, Failed:  0, Skipped: 0, Time:   0,098s
       baseservices.threading.XUnitWrapper                      Total:    7, Errors: 0, Failed:  0, Skipped: 0, Time:   8,176s
       baseservices.TieredCompilation.XUnitWrapper              Total:   12, Errors: 0, Failed:  0, Skipped: 0, Time:   9,758s
       baseservices.typeequivalence.XUnitWrapper                Total:    1, Errors: 0, Failed:  1, Skipped: 0, Time:   0,169s
       baseservices.varargs.XUnitWrapper                        Total:    2, Errors: 0, Failed:  0, Skipped: 0, Time:   0,152s
       CoreMangLib.system.XUnitWrapper                          Total:    1, Errors: 0, Failed:  0, Skipped: 0, Time:   0,133s
       Exceptions.ForeignThread.XUnitWrapper                    Total:    1, Errors: 0, Failed:  0, Skipped: 0, Time:   0,138s
       GC.API.XUnitWrapper                                      Total:   21, Errors: 0, Failed:  0, Skipped: 0, Time:   9,426s
       GC.Coverage.XUnitWrapper                                 Total:    2, Errors: 0, Failed:  0, Skipped: 0, Time:   0,968s
       GC.Features.XUnitWrapper                                 Total:   19, Errors: 0, Failed:  0, Skipped: 0, Time:  18,081s
       GC.LargeMemory.XUnitWrapper                              Total:    3, Errors: 0, Failed:  0, Skipped: 0, Time:   0,188s
       GC.Regressions.XUnitWrapper                              Total:   10, Errors: 0, Failed:  0, Skipped: 0, Time:   4,215s
       GC.Scenarios.XUnitWrapper                                Total:   32, Errors: 0, Failed:  2, Skipped: 0, Time:   7,551s
       GC.Stress.XUnitWrapper                                   Total:    1, Errors: 0, Failed:  0, Skipped: 0, Time:   0,040s
       ilasm.PortablePdb.XUnitWrapper                           Total:    1, Errors: 0, Failed:  0, Skipped: 0, Time:   0,687s
       ilasm.System.XUnitWrapper                                Total:    1, Errors: 0, Failed:  0, Skipped: 0, Time:   0,241s
       Interop.ArrayMarshalling.XUnitWrapper                    Total:    1, Errors: 0, Failed:  1, Skipped: 0, Time:   0,161s
       Interop.DllImportAttribute.XUnitWrapper.dll              Total:    0
       Interop.ExecInDefAppDom.XUnitWrapper                     Total:    1, Errors: 0, Failed:  1, Skipped: 0, Time:   0,128s
       Interop.ICastable.XUnitWrapper.dll                       Total:    0
       Interop.ICustomMarshaler.XUnitWrapper                    Total:    2, Errors: 0, Failed:  1, Skipped: 0, Time:   2,364s
       Interop.IJW.XUnitWrapper                                 Total:    6, Errors: 0, Failed:  6, Skipped: 0, Time:   2,010s
       Interop.LayoutClass.XUnitWrapper.dll                     Total:    0
       Interop.MarshalAPI.XUnitWrapper                          Total:    2, Errors: 0, Failed:  2, Skipped: 0, Time:   0,189s
       Interop.NativeLibrary.XUnitWrapper                       Total:    3, Errors: 0, Failed:  2, Skipped: 0, Time:   2,376s
       Interop.PInvoke.XUnitWrapper                             Total:   66, Errors: 0, Failed: 11, Skipped: 0, Time:   8,776s
       Interop.StringMarshalling.XUnitWrapper                   Total:    1, Errors: 0, Failed:  1, Skipped: 0, Time:   0,171s
       Interop.StructMarshalling.XUnitWrapper                   Total:    3, Errors: 0, Failed:  2, Skipped: 0, Time:   0,196s
       Interop.StructPacking.XUnitWrapper.dll                   Total:    0
       Interop.SuppressGCTransition.XUnitWrapper                Total:    1, Errors: 0, Failed:  0, Skipped: 0, Time:   0,139s
       Interop.UnmanagedCallConv.XUnitWrapper                   Total:    1, Errors: 0, Failed:  0, Skipped: 0, Time:   0,132s
       Interop.UnmanagedCallersOnly.XUnitWrapper                Total:    1, Errors: 0, Failed:  1, Skipped: 0, Time:   0,194s
       Interop.WinRT.XUnitWrapper                               Total:    1, Errors: 0, Failed:  1, Skipped: 0, Time:   0,161s
       JIT.CheckProjects.XUnitWrapper                           Total:    1, Errors: 0, Failed:  0, Skipped: 0, Time:   0,126s
       JIT.CodeGenBringUpTests.XUnitWrapper                     Total:   32, Errors: 0, Failed:  0, Skipped: 0, Time:   0,790s
       JIT.Directed.XUnitWrapper                                Total:  138, Errors: 0, Failed:  6, Skipped: 0, Time:   4,566s
       JIT.Generics.XUnitWrapper                                Total:   42, Errors: 0, Failed:  0, Skipped: 0, Time:   0,954s
       JIT.HardwareIntrinsics.XUnitWrapper                      Total:  354, Errors: 0, Failed:  0, Skipped: 0, Time:  10,217s
       JIT.IL_Conformance.XUnitWrapper                          Total:  112, Errors: 0, Failed:  0, Skipped: 0, Time:   2,606s
       JIT.Intrinsics.XUnitWrapper                              Total:   24, Errors: 0, Failed:  0, Skipped: 0, Time:   0,738s
       JIT.jit64.XUnitWrapper                                   Total:   89, Errors: 0, Failed: 14, Skipped: 0, Time:  29,256s
       JIT.Math.XUnitWrapper                                    Total:    2, Errors: 0, Failed:  0, Skipped: 0, Time:   0,140s
       JIT.Methodical.XUnitWrapper                              Total:  757, Errors: 0, Failed:  2, Skipped: 0, Time:  17,282s
       JIT.opt.XUnitWrapper                                     Total:   81, Errors: 0, Failed:  0, Skipped: 0, Time:   2,736s
       JIT.Performance.XUnitWrapper                             Total:   84, Errors: 0, Failed:  0, Skipped: 0, Time: 101,999s
       JIT.Regression.XUnitWrapper                              Total:  665, Errors: 0, Failed:  1, Skipped: 0, Time:  16,468s
       JIT.RyuJIT.XUnitWrapper                                  Total:    1, Errors: 0, Failed:  0, Skipped: 0, Time:   0,142s
       JIT.SIMD.XUnitWrapper                                    Total:   99, Errors: 0, Failed:  0, Skipped: 0, Time:   2,709s
       JIT.Stress.XUnitWrapper.dll                              Total:    0
       JIT.superpmi.XUnitWrapper.dll                            Total:    0
       Loader.AssemblyDependencyResolver.XUnitWrapper.dll       Total:    0
       Loader.AssemblyLoadContext30Extensions.XUnitWrapper.dll  Total:    0
       Loader.binding.XUnitWrapper.dll                          Total:    0
       Loader.classloader.XUnitWrapper                          Total:   75, Errors: 0, Failed:  0, Skipped: 0, Time:  12,806s
       Loader.CollectibleAssemblies.XUnitWrapper.dll            Total:    0
       Loader.ContextualReflection.XUnitWrapper.dll             Total:    0
       Loader.regressions.XUnitWrapper                          Total:    1, Errors: 0, Failed:  0, Skipped: 0, Time:   0,135s
       managed.Compilation.XUnitWrapper                         Total:    1, Errors: 0, Failed:  0, Skipped: 0, Time:   0,895s
       profiler.elt.XUnitWrapper.dll                            Total:    0
       profiler.eventpipe.XUnitWrapper.dll                      Total:    0
       profiler.gc.XUnitWrapper.dll                             Total:    0
       profiler.multiple.XUnitWrapper.dll                       Total:    0
       profiler.rejit.XUnitWrapper.dll                          Total:    0
       profiler.transitions.XUnitWrapper.dll                    Total:    0
       profiler.unittest.XUnitWrapper.dll                       Total:    0
       readytorun.crossgen2.XUnitWrapper.dll                    Total:    0
       readytorun.DynamicMethodGCStress.XUnitWrapper.dll        Total:    0
       readytorun.multifolder.XUnitWrapper.dll                  Total:    0
       readytorun.r2rdump.XUnitWrapper.dll                      Total:    0
       readytorun.tests.XUnitWrapper.dll                        Total:    0
       reflection.DefaultInterfaceMethods.XUnitWrapper          Total:    1, Errors: 0, Failed:  0, Skipped: 0, Time:   0,132s
       reflection.GenericAttribute.XUnitWrapper.dll             Total:    0
       reflection.Modifiers.XUnitWrapper.dll                    Total:    0
       reflection.RefEmit.XUnitWrapper                          Total:    1, Errors: 0, Failed:  0, Skipped: 0, Time:   0,172s
       reflection.regression.XUnitWrapper                       Total:    1, Errors: 0, Failed:  0, Skipped: 0, Time:   0,122s
       reflection.SetValue.XUnitWrapper.dll                     Total:    0
       reflection.StaticInterfaceMembers.XUnitWrapper           Total:    1, Errors: 0, Failed:  0, Skipped: 0, Time:   0,123s
       Regressions.coreclr.XUnitWrapper                         Total:   17, Errors: 0, Failed:  0, Skipped: 0, Time:   0,342s
       tracing.eventactivityidcontrol.XUnitWrapper              Total:    1, Errors: 0, Failed:  0, Skipped: 0, Time:   0,138s
       tracing.eventcounter.XUnitWrapper                        Total:    8, Errors: 0, Failed:  0, Skipped: 0, Time:   0,072s
       tracing.eventlistener.XUnitWrapper.dll                   Total:    0
       tracing.eventpipe.XUnitWrapper                           Total:   10, Errors: 0, Failed:  0, Skipped: 0, Time:   0,334s
                                                                       ----          -          --           -        --------
                                                          GRAND TOTAL: 2804          0          55           0        283,248s (284,949s)
```

So we run 2804 runtime tests on Mono Windows and we currently have 55 failures. I briefly looked at several of the failures and many of them are either running things not supported on Mono, but since its Windows it will still try to execute some of these since they where never explicitly excluded on Mono (just relying on the fact that they are Windows specific and won't run on other platforms). There are also some tests failing due to issues resolving a dll import using ordinal, pretty much all 14 failures in JIT.jit64.XUnitWrapper fails due to that.

I will probably take a look at this in main and see if I we can either fix or explicitly exclude these tests on Mono. Once that is done I believe we should aim at getting the runtime tests CI lane running on Mono Windows in main.

**Comment by lateralusX** [comment]
https://github.com/dotnet/runtime/pull/64281

**Comment by lateralusX** [comment]
https://github.com/dotnet/runtime/pull/64281 is now merged.
